File: src/client/NodeJS/ms-rest/lib/msRest.js
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

var utils = require('./utils');
exports.Constants = require('./constants');

exports.WebResource = require('./webResource');
exports.ServiceClient = require('./serviceClient');
exports.HttpOperationResponse = require('./httpOperationResponse');

// Credentials
exports.TokenCredentials = require('./credentials/tokenCredentials');
exports.BasicAuthenticationCredentials = require('./credentials/basicAuthenticationCredentials');

// Other filters
exports.ProxyFilter = require('./filters/proxyFilter');
exports.LogFilter = require('./filters/logFilter');
exports.SigningFilter = require('./filters/signingFilter');
exports.ExponentialRetryPolicyFilter = require('./filters/exponentialRetryPolicyFilter');

exports.requestPipeline = require('./requestPipeline');
exports.stripResponse = utils.stripResponse;
exports.stripRequest = utils.stripRequest;
exports.isValidUuid = utils.isValidUuid;

//serialization
exports.serializeObject = require('./serialization').serializeObject;
exports.serialize = require('./serialization').serialize;
exports.deserialize = require('./serialization').deserialize;
var serialization = require('./serialization');

exports.addSerializationMixin = function (destObject) {
  ['serialize', 'serializeObject', 'deserialize'].forEach(function (property) {
    destObject[property] = serialization[property];
  });
};

exports = module.exports;

File: src/client/Ruby/ms-rest/lib/ms_rest.rb
# encoding: utf-8
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See License.txt in the project root for license information.

require 'base64'
require 'openssl'
require 'faraday'
require 'timeliness'
require 'ms_rest/version'

require 'ms_rest/credentials/token_provider'
require 'ms_rest/credentials/string_token_provider'
require 'ms_rest/credentials/service_client_credentials'
require 'ms_rest/credentials/basic_authentication_credentials'
require 'ms_rest/credentials/token_credentials'

require 'ms_rest/rest_error.rb'
require 'ms_rest/deserialization_error.rb'
require 'ms_rest/validation_error.rb'
require 'ms_rest/serialization.rb'
require 'ms_rest/http_operation_response'
require 'ms_rest/http_operation_request'
require 'ms_rest/http_operation_error'
require 'ms_rest/retry_policy_middleware'
require 'ms_rest/service_client'

module MsRest end
module MsRest::Serialization end
